### DISCUSSION TITLE
Espresso Current JDK Support?

### DISCUSSION THREAD

**Discussion by wilgaboury** [oracle/graal#8812] [Q&A] [answered]
On the readme page found at https://github.com/oracle/graal/tree/master/espresso, it says "Espresso is a complete Java implementation which is kept up to date with the latest LTS JDK versions", but in a bunch of other assorted documentation it claims that espresso only supports JDK 8 and 11. I would like to use espresso for a project on JDK 21, is that currently possible?

**Comment by mukel** [comment, marked as answer]
Espresso requires a 21+ host, but can spawn guest contexts of Java 8, 11, 17 and 21 e.g. `--java.JavaHome=/path/to/guest/java/home` e.g. it will shape-shift into the version of specified guest Java home.
If you don't specify the guest Java home, it will try to use the same as the host; so yes, you can use Espresso to run a Java 21 guest context on a Java 21 host.
